### PR TITLE
docs(upgrading): add upgrade guide for v2.18.0

### PR DESCRIPTION
**Pull request by astencel-sumo**
Upgrading to this release with the OpenTelemetry Operator enabled breaks unless you upgrade the operator's CRDs. This needs to be documented.

### CODE REVIEW

File: deploy/docs/v2-18-migration.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+### Do I need to do this?
+
+You need to do this if you have the OpenTelemetry Operator subchart enabled with `opentelemetry-operator.enabled: true` in your values file.
+
+### What do I need to do?
+
+Run the following commands on the cluster before upgrading the chart:
+
+```shell
+kubectl apply --server-side -f https://raw.githubusercontent.com/open-telemetry/opentelemetry-helm-charts/opentelemetry-operator-0.24.0/charts/opentelemetry-operator/crds/crd-opentelemetrycollector.yaml
```
Comment (rnishtala-sumo) [inline deploy/docs/v2-18-migration.md:14]: do we need the `--force-conflicts` option like here:
https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/ba08b58707e58cdb50da0032f0be1e4c8997f0dc/docs/v3-migration-doc.md#upgrade-kube-prometheus-stack

Comment (rnishtala-sumo) [reply deploy/docs/v2-18-migration.md:14]: oh nvm, I see that we have `-f`

**Review by rnishtala-sumo — APPROVED** [review]

File: deploy/docs/v2-18-migration.md (hunk 1 above)
Comment (astencel-sumo) [reply deploy/docs/v2-18-migration.md:14]: It's a good point that the flag should be probably in long format, I'll change that :+1:

Comment (astencel-sumo) [reply deploy/docs/v2-18-migration.md]: Added the flag `--force-conflicts`.

**Pull request merged**